pyqtgraph/parametertree/parameterTypes.py

```diff
@@ -681,3 +682,241 @@ class TextParameter(Parameter):
+    def makeWidget(self):
+        w = QtGui.QCalendarWidget()
+        w.setMaximumHeight(200)
+        w.sigChanged = w.selectionChanged
+        w.value = w.selectedDate
+        w.setValue = w.setSelectedDate
+        self.widget = w
+        self.hideWidget = False
+        self.param.opts['default'] = QtCore.QDate.currentDate()
+        return w
+
+class CalendarParameter(Parameter):
+    itemClass = CalendarParameterItem
+
+registerParameterType('calendar', CalendarParameter, override=True)
+
+
+
+class PenParameterItem(ParameterItem):
+    def __init__(self, param, depth):
+        ParameterItem.__init__(self, param, depth)
+        self.layoutWidget = QtGui.QWidget()
+        self.layout = QtGui.QHBoxLayout()
+        self.layoutWidget.setLayout(self.layout)
+        self.button = QtGui.QPushButton()
+        #larger button
+        self.button.setFixedWidth(100)
+        self.layout.addWidget(self.button)
+        self.layout.addStretch()
+        self.pen = QtGui.QPen()
+        self.button.clicked.connect(self.buttonClicked)
+        #clear button for drawing, override paint event
+        self.setText(0, '')
+        self.button.paintEvent = self.buttonPaintEvent
+        
+
+    def value(self):
+        return self.pen
+
+    def setValue(self,pen):
+        self.penChanged(pen)
+
+    def treeWidgetChanged(self):
+        ParameterItem.treeWidgetChanged(self)
+        tree = self.treeWidget()
+        if tree is None:
+            return
+
+        tree.setFirstItemColumnSpanned(self, True)
+        tree.setItemWidget(self, 0, self.layoutWidget)
+
+    def buttonClicked(self):
+        #open up the pen selector dialog
+        self.oldPen = fn.mkPen(self.pen)
+        self.pdialog = PenSelectorDialog(fn.mkPen(self.pen),QtGui.QApplication.activeWindow())
+        self.pdialog.penChanged.connect(self.penChanged)
+        self.pdialog.finished.connect(self.penChangeFinished)
+        self.pdialog.exec_()
+
+    def penChangeFinished(self,ret):
+        #finished changing
+        if not ret:
+            #revert if cancel
+            self.penChanged(self.oldPen)
+        else:
+            #event if accepted
+            self.param.penChanged(self.pen)
+
+    def penChanged(self,pen):
+        if not isinstance(pen,QtGui.QPen):
+            pen = fn.mkPen(pen)
+        pen.setCosmetic(True)
+        self.pen = pen
+
+    def buttonPaintEvent(self, event):
+        #draw a button as usual
+        QtGui.QPushButton.paintEvent(self.button, event)
+
+        path = QtGui.QPainterPath()
+        displaySize = self.button.size()
+        labelBackgroundColor = self.button.palette().background().color()
+        w,h = displaySize.width(),displaySize.height()
+        #draw a squiggle with the pen
+        path.moveTo(w*.2,h*.2)
+        path.lineTo(w*.4,h*.8)
+        path.cubicTo(w*.5,h*.1,w*.7,h*.1,w*.8,h*.8)
+
+        painter = QtGui.QPainter(self.button)
+        painter.setPen(self.pen)
```

Comment:
This uses the current value of the parameter tree to draw the squiggle on the button
```suggestion
        painter.setPen(self.param.value())
```